packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -60,7 +64,17 @@ class SharedPreferences {
 
   /// Reads a set of string values from persistent storage,
   /// throwing an exception if it's not a string set.
-  List<String> getStringList(String key) => _preferenceCache[key] as List<String>;
+  List<String> getStringList(String key) =>
+      _preferenceCache[key] as List<String>;
+
+  /// Gets a JSON-encoded preference string and returns it as an Object. See
+  /// [JSON.decode] and [JsonDecoder] documentation for what happens when the
+  /// string is not a valid JSON string, and how to use the reviver.  Returns
+  /// null if the given key does not exist.
+  Object getObject(String key, {reviver(var key, var value)}) {
+    if (_preferenceCache[key] == null) return null;
```

Comment:
Sounds good.

Yes, I believe for storing large structured objects an async local storage API is better suited. One candidate is SQLite: https://github.com/tekartik/sqflite